### PR TITLE
YAPF -> Black, drop py36 and `async_generator`, test on newer Pythons

### PR DESCRIPTION
**Pull request by Zac-HD**
This is intended as a base for the actual feature work of #104 and #128; as a separate PR to make the next one much easier to review... and ensure that the tests all pass with just this base 😅 

I also created https://github.com/python-trio/trio/issues/2458 to track `async_generator` upstream.

### CODE REVIEW

**Comment by Zac-HD** [comment]
I'm not sure what's happened to `codecov`, but the other "expected checks" are removed-and-replaced in this PR; if it otherwise looks good we'll need to adjust the branch protection settings.

File: pytest.ini [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [pytest]
-addopts = -W error -ra -v --pyargs pytest_trio --verbose --cov
```
Comment (pquentin) [inline pytest.ini:2]: What was the issue with -W error?

**Review by pquentin — APPROVED** [review]
Thanks! This looks good to me.

**Comment by pquentin** [comment]
codecov is indeed broken, we're sending empty reports: https://app.codecov.io/github/python-trio/pytest-trio/commit/f788fe7d1bc7e7cbfa8e86d482018afd785536d1. The uploader complains about the checkout, maybe that's the issue.

I update the required checks so this can now be merged!

File: pytest.ini (hunk 1 above)
Comment (Zac-HD) [reply pytest.ini:2]: MultiError is deprecated, and I didn't want to go for elaborate workarounds when I've got another PR ready to fix it (and re-enable -W error) so soon.

**Pull request merged**